### ISSUE TITLE
Improve RankD Tests

### ISSUE THREAD

**Issue by bbengfort** [DistrictDataLabs/yellowbrick#435]
Right now the Rank1D and Rank2D tests are very basic and can be improved using the new assert image similarity and pytest testing framework mechanisms. 

### Proposal/Issue

The test matrix should against the following:

 - [x] replace `make_regression` dataset with `load_energy` and update those tests
 - [x] use `load_occupancy` for classification tests
 - [x] algorithms: pearson, covariance, spearman (2D) and shaprio (1D)
 - [x] 1D case - horizontal orientation
 - [x] 1D case - vertical orientation
 - [x] Test that an exception is raised for unrecognized algorithms
 - [x] test the underlying rank matrix is correct

Unfortunately, we can't use `pytest.mark.parametrize` with visual test cases (yet), so we'll have to make individual tests for each. 

### Code Snippet

Tests will look approximately like:

```python
def test_rank2d_bad_algorithm(self):
    """"
    Assert that unknown algorithms raise exception
    """"
    with pytest.raises(YellowbrickValueError, match="unknown algorithm"):
         # do the thing 

def test_rank2d_pearson_regression(self):
    """"
    Test Rank2D images similar with pearson scores on regression dataset 
    """"
    data = load_energy(return_dataset=True)
    oz = Rank2D(algorithm='pearson')
    oz.fit_transform(data) 
    npt.assert_array_equal(oz.ranks_, [[]])
    self.assert_images_similar(oz, tol=0.25) 
```

### Background

See #68 and #429

**Comment by tabishsada** [comment]
I'll work on this.

**Comment by yanigisawa** [comment]
Are there additional tests / algorithms expected here beyond the Kendal-Tau metric added above?

**Comment by lwgray** [comment]
@yanigisawa Thanks for the comment. However, The Kendall-tau metric was already added in PR #645
